addons/account/models/account_tax.py

```diff
@@ -225,6 +238,84 @@ def _compute_tax_group_id(self):
                 ('country_id', '=', False),
             ], limit=1)
 
+    def _hook_compute_is_used(self):
+        '''
+            To be overriden to add taxed transactions in the computation of `is_used`
+            Should return a Counter containing a dictionary {record: int} where
+            the record is an account.tax object. The int should be greater than 0
+            if the tax is used in a transaction.
+        '''
+        return Counter()
+
+    def _compute_is_used(self):
+        taxes_in_transactions_ctr = (
+            Counter(dict(self.env['account.move.line']._read_group([], groupby=['tax_ids'], aggregates=['__count']))) +
+            Counter(dict(self.env['account.reconcile.model.line']._read_group([], groupby=['tax_ids'], aggregates=['__count']))) +
+            self._hook_compute_is_used()
+        )
+        for tax in self:
+            tax.is_used = bool(taxes_in_transactions_ctr[tax])
+
+    @api.depends('repartition_line_ids.account_id', 'repartition_line_ids.factor_percent', 'repartition_line_ids.use_in_tax_closing', 'repartition_line_ids.tag_ids')
+    def _compute_repartition_lines_str(self):
+        for tax in self:
+            repartition_lines_str = tax.repartition_lines_str or ""
+            if tax.is_used:
+                for repartition_line in tax.repartition_line_ids:
+                    repartition_line_info = {
+                        'id': repartition_line.id,
+                        'Factor Percent': repartition_line.factor_percent,
+                        'Account': repartition_line.account_id.name or 'None',
+                        'Tax Grids': repartition_line.tag_ids.mapped('name') or 'None',
+                        'Use in tax closing': str(repartition_line.use_in_tax_closing),
+                    }
+                    repartition_lines_str += str(repartition_line_info) + '//'
+                repartition_lines_str = repartition_lines_str.strip('//')
+            tax.repartition_lines_str = repartition_lines_str
+
+    def _message_log_repartition_lines(self, old_value_str, new_value_str):
+        old_values = old_value_str.split('//')
+        new_values = new_value_str.split('//')
+        if len(old_values) != len(new_values):
+            raise UserError(_("Taxes should have the same amount of invoice and refund repartition lines."))
```

Comment:
{old/new}_values is a list in which each item is a repartition line. The error message should in fact be more like: """This tax has already been used, you cannot add nor delete its tax repartition lines."""
This code is only meant to be executed when the tax is used (called from _message_log). I have added a check in case someone else decides (for any reason) to call the function from elsewhere.

Reply:
but this error message already exists, in create() and unlink() of tax.repartition.line...

Reply:
What do you think I should be doing if the number of repartition line differs after the modification of a used tax? Should I simply do nothing and return the function?

Reply:
or simply not check at all and assume that the number will always be the same since a tax that is used does not allow for lines to be added and/or removed?

Reply:
> simply not check at all and assume that the number will always be the same since a tax that is used does not allow for lines to be added and/or removed?

yes indeed... I don't see how that could happen, so I don't see the purpose of checking that :)

Reply:
Removed